### ISSUE TITLE
Pytest runner needs to not count skipped tests as failures

### ISSUE THREAD

**Issue by abingham** [sixty-north/cosmic-ray#184]
Right now the pytest-runner only counts "passed" tests as non-failures, but it also needs to look for skipped tests. There might be other categories of non-failure that we need to account for.

**Comment by grapherd** [comment]
We can filter collector report before return the result back

```python
# Skip skipped test
collector.reports = filter(lambda rep: not rep.skipped, collector.reports)
return (
    all(not r.failed for r in collector.reports),
    [repr(r) for r in collector.reports if r.failed or r.skipped])
```

Not sure if this is workable.

Also, is CR driven by source file but not unit-test?
If the test class (e.g. `class foo(unittest.TestCase`) was marked as skipped, CR will still generate corresponding mutant cases, it seem we can somehow skip this?

**Comment by abingham** [comment]
Oops, this has already been addressed in 40c272ee66545691734642d766b01c8659fcd659. I should have closed it.

**Comment by abingham** [comment]
> Also, is CR driven by source file but not unit-test?
> If the test class (e.g. class foo(unittest.TestCase) was marked as skipped, CR will still generate > corresponding mutant cases, it seem we can somehow skip this?

I'm not sure what you mean by this. CR never looks at the tests themselves, and it doesn't really care if tests are marked as skipped or not. We never mutate the tests, only the "production" code, so we'll generate mutants independent of whether any test is marked as skipped. 

In fact, it's very important that CR generates mutants in the presence of skipped tests because mutation testing may show that those tests need to be re-enabled.
